com.ibm.wala.core/src/com/ibm/wala/ssa/DefUse.java

```diff
@@ -187,4 +187,14 @@ public void remove() {
   public int getNumberOfUses(int v) {
     return uses[v] == null ? 0 : uses[v].size();
   }
+
+  /**
+   * @param v a value number
+   * @return true if the variable with the given value number has no uses
+   */
+  public boolean isUnused(int v) {
+    final MutableIntSet usesSet = uses[v];
+    assert usesSet == null || !usesSet.isEmpty();
```

Comment:
Seems like an odd place for this assertion?  If we care we should assert at the point where we construct the `DefUse`, for all value numbers

Reply:
In a build with assertions off, the code in this method simplifies to `return uses[v] == null`. Quite efficient, but wrong if `uses[v]` were present but empty. I put this assertion here because here is where I rely on it being true.

I am happy to broaden the check to all value numbers and move it to the end of the `DefUse` constructor. I’m equally happy to leave the check here or to remove the check entirely. Please let me know which you like best and I will make it so.